faros-airbyte-cdk/src/prompts.ts

```
@@ -39,9 +45,24 @@ export function runSelect(cfg: SelectConfig): Promise<string> {
       if (choices.has(choice.type)) return;
       choices.set(choice.type, choice);
     });
-    // Return the first choice of the lowest order
+    /**
+     * Return the first choice of the lowest order for autofill.
+     * Check `ChoiceType` enum: SKIP -> DEFAULT -> EXAMPLE -> ENVIRONMENT_VARIABLE -> BOOLEAN -> ENUM -> USER_INPUT
+     *
+     * If the config is not requried, SKIP choice is pushed before this function is called and makes the config not filled.
+     * If the config falls into the last choice `USER_INPUT`, it will be autofilled with the placeholder value.
+     *
+     * If the config is a leaf, skip EXAMPLE and ENUM.
+     * That means we don't autofill EXAMPLE and ENUM values as that might get confused with DEFAULT values.
+     */
     for (const symbol of Object.values(ChoiceType)) {
-      if (choices.has(symbol)) {
+      if (
+        choices.has(symbol) &&
+        !(
+          cfg.name === SelectConfigName.LEAF &&
+          (symbol === ChoiceType.EXAMPLE || symbol === ChoiceType.ENUM)
```

@cjwooo  i want to not autofill examples and enums when there's no default for the configs. i think it will be confused with the defaults.

If the config doesn't have default, then it will fall back to placeholders. for `PLACEHOLDER_NUMBER` and `PLACEHOLDER_BOOLEAN`, it might still be confusing but i don't know if there a better way to get around that.